src/bms/ast/ast_extract.rs

```diff
@@ -12,13 +12,21 @@ pub(super) fn extract_units<'a>(
     for unit in units {
         match unit {
             Unit::TokenWithPos(token) => {
-                tokens.push(token.clone());
+                tokens.push((*token).clone());
```

Comment:
```suggestion
                tokens.push(token.clone());
```
This change is needless due to auto-deref.